test/test_encryption.py

```diff
@@ -2687,5 +2687,162 @@ def test_int(self):
+            OperationFailure,
+            "create.encryptedFields.fields.keyId' is the wrong type 'bool', expected type 'binData",
+        ):
+            self.client_encryption.create_encrypted_collection(
+                database=self.db,
+                name="testing1",
+                encryptedFields={"fields": [{"path": "ssn", "bsonType": "string", "keyId": False}]},
+                kms_provider="local",
+            )
+
+    def test_04_insert_encrypted(self):
+        coll, ef = self.client_encryption.create_encrypted_collection(
+            database=self.db,
+            name="testing1",
+            encryptedFields={"fields": [{"path": "ssn", "bsonType": "string", "keyId": None}]},
+            kms_provider="local",
+        )
+        key1_id = ef["fields"][0]["keyId"]
+        encrypted_value = self.client_encryption.encrypt(
+            "123-45-6789",
+            key_id=key1_id,
+            algorithm=Algorithm.UNINDEXED,
+        )
+        coll.insert_one({"ssn": encrypted_value})
+
+    def test_copy_encrypted_fields(self):
+        encrypted_fields = {
+            f"{self.db.name}.testing1": {
+                "fields": [
+                    {
+                        "path": "ssn",
+                        "bsonType": "string",
+                        "keyId": self.client_encryption.create_data_key(kms_provider="local"),
+                    }
+                ]
+            }
+        }
+        opts = AutoEncryptionOpts(
+            {"local": {"key": LOCAL_MASTER_KEY}},
+            self.key_vault.full_name,
+            bypass_query_analysis=True,
+            encrypted_fields_map=encrypted_fields,
+        )
+        client = rs_or_single_client(auto_encryption_opts=opts)
+        _, ef = self.client_encryption.create_encrypted_collection(
+            database=client.db, name="testing1", kms_provider="local"
+        )
+        self.assertIsNot(
+            ef,
+            client.options.auto_encryption_opts._encrypted_fields_map[f"{self.db.name}.testing1"],
+        )
+
+    def test_options_forward(self):
+        coll, ef = self.client_encryption.create_encrypted_collection(
+            database=self.db,
+            name="testing1",
+            kms_provider="local",
+            encryptedFields={"fields": [{"path": "ssn", "bsonType": "string", "keyId": None}]},
+            read_preference=ReadPreference.NEAREST,
+        )
+        self.assertEqual(coll.read_preference, ReadPreference.NEAREST)
+        self.assertEqual(coll.name, "testing1")
+
+    def test_mixed_null_keyids(self):
+        key = self.client_encryption.create_data_key(kms_provider="local")
+        coll, ef = self.client_encryption.create_encrypted_collection(
+            database=self.db,
+            name="testing1",
+            encryptedFields={
+                "fields": [
+                    {"path": "ssn", "bsonType": "string", "keyId": None},
+                    {"path": "dob", "bsonType": "string", "keyId": key},
```

Comment:
Can we add a field that omits keyId?

Reply:
Done.

Reply:
Not done. Both these fields have `"keyId": None` but we should test when the "keyId" field is missing (not None).

Reply:
Ah, that makes sense. Fixed.

Reply:
This is still not fixed.

Reply:
The next test contains the test case you mentioned--line 2828

Reply:
I will move it to this test because I think it makes more sense here. I also made the test use all of the fields that I added.

Reply:
Ack.